azurerm/internal/services/applicationinsights/application_insights_smart_detection_rule_resource.go

```diff
@@ -77,10 +71,7 @@ func resourceApplicationInsightsSmartDetectionRuleUpdate(d *pluginsdk.ResourceDa
 	defer cancel()
 
 	log.Printf("[INFO] preparing arguments for AzureRM Application Insights Samrt Detection Rule update.")
-
-	// The Smart Detection Rule name from the UI doesn't match what the API accepts.
-	// We'll have the user submit what the name looks like in the UI and trim it behind the scenes to match what the API accepts
-	name := strings.ToLower(strings.Join(strings.Split(d.Get("name").(string), " "), ""))
+	name := strings.Join(strings.Split(strings.ToLower(d.Get("name").(string)), " "), "")
```

Comment:
this is a breaking change so we can't change this behaviour till 3.0, could we add a comment to that affect?